modules/installer.nix

```diff
@@ -35,6 +39,132 @@ with builtins; with lib; {
+          stage2 = pkgs.writeScript "updater-stage-2.sh" ''
+            #!${busybox}/bin/sh
+            set -euo pipefail
+
+            # Set PATH to include busybox
+            export PATH=${busybox}/bin:$PATH
+
+            # Root check
+            if [ `id -u` != "0" ]; then
+              echo "This script must be run as root" >&2
+              exit 1
+            fi
+
+
+            function prompt_yn {
+              while true; do
+                read -p "$@ [y/n] " result
+                case $result in
+                  [Yy]* ) return 0;;
+                  [Nn]* ) return 1;;
+                  * ) echo "Please answer yes or no.";;
+                esac
+              done
+            }
+
+            #Parse arguments
+            GC=true
+            REBUILD=false
+            for arg in "$@"; do
+              case $arg in
+                --no-gc)
+                  GC=false
+                  ;;
+                --rebuild)
+                  REBUILD=true
+                  ;;
+                *)
+                  echo "Unknown argument: $arg"
+                  exit 1
+                  ;;
+              esac
+            done
+
+
+            # Check if /etc/nixos/nixos-wsl exists
+            if ! [ -d /etc/nixos/nixos-wsl ]; then
+              echo "/etc/nixos/nixos-wsl was not found! Can not update this system" >&2
+              exit 1
+            fi
+
+            # Unpack rootfs
+            echo "Unpacking..."
+            mkdir -p /tmp/nixos-wsl-updater/rootfs
+            ${pv}/bin/pv ${rootfs} | tar -C /tmp/nixos-wsl-updater/rootfs -xz
+
+            # Detect flake.nix
+            FLAKE=false
+            if [ -f /etc/nixos/flake.nix ]; then
+              echo "Flake detected!"
+              echo "Please update your nixpkgs input to \"github:nixos/nixpkgs/nixos-${config.system.nixos.release}\" and run nixos-rebuild to complete the update"
+              echo "Press enter to continue..."
+              read
+              FLAKE=true
+            fi
+
+            # Check that nixos-rebuild builds the current configuration
+            EXTERNAL=false
+            if ! $FLAKE; then
+              dir=$PWD
+              cd $(mktemp -d)
+              nixos-rebuild build
+              if [[ $(readlink /run/current-system) != $(readlink result) ]]; then
+                echo "The current configuration does not match what is produced by nixos-rebuild!" >&2
+                echo "Please make sure to update your nixos channel or nixpkgs input to NixOS ${config.system.nixos.release}" >&2
+                echo "Press enter to continue..."
+                read
+                EXTERNAL=true
+              fi
+              cd $dir
+            fi
+
+            # Update channels
+            if ! $FLAKE && ! $EXTERNAL; then
+              echo "Updating channels..."
+              nix-channel --add https://nixos.org/channels/nixos-${config.system.nixos.release} nixos
+              nix-channel --update -v
```

Comment:
If we would add this to the installer it would not rely on the network, but also be bigger.

Reply:
Given that this is for updating, it should always give you the latest nixpkgs. Trying to update without access to the internet would break at nixos-rebuild anyway

Reply:
Can't we just instruct people to manually do the update?
Most releases we have deprecations that people should handle anyway and I suspect this to open another pandoras box.

Reply:
Having people try to run this on NixOS versions that are older than what we test against might lead to new and interesting bugs that I don't want to deal with. There should probably be a check for nixos-unstable though